link/src/main/java/com/stripe/android/link/ui/paymentmethod/SupportedPaymentMethod.kt

```diff
@@ -11,58 +15,58 @@ import com.stripe.android.ui.core.forms.LinkCardForm
+ * @param primaryButtonEndIconResourceId Drawable resource id for the icon to be displayed at the
+ *      end of the primary button when this payment method is being created.
  */
 internal enum class SupportedPaymentMethod(
     val type: String,
-    val formSpec: List<FormItemSpec>
+    val formSpec: List<FormItemSpec>,
+    @StringRes val nameResourceId: Int,
+    @DrawableRes val iconResourceId: Int,
+    @DrawableRes val primaryButtonStartIconResourceId: Int? = null,
+    @DrawableRes val primaryButtonEndIconResourceId: Int? = null
 ) {
     Card(
         ConsumerPaymentDetails.Card.type,
-        LinkCardForm.items
+        LinkCardForm.items,
+        R.string.stripe_paymentsheet_payment_method_card,
+        R.drawable.ic_link_card,
+        primaryButtonEndIconResourceId = R.drawable.stripe_ic_lock
     ) {
-        override fun createParams(
-            paymentMethodCreateParams: PaymentMethodCreateParams,
-            email: String
-        ) = ConsumerPaymentDetailsCreateParams.Card(
-            paymentMethodCreateParams.toParamMap(),
-            email
-        )
-
-        /**
-         * CVC is not passed during creation, and must be included when confirming the payment.
-         */
-        override fun extraConfirmationParams(paymentMethodCreateParams: PaymentMethodCreateParams) =
-            (paymentMethodCreateParams.toParamMap()["card"] as? Map<*, *>)?.let { card ->
-                mapOf("card" to mapOf("cvc" to card["cvc"]))
-            }
+        override fun primaryButtonLabel(
+            stripeIntent: StripeIntent,
+            resources: Resources
+        ) = completePaymentButtonLabel(stripeIntent, resources)
     },
     BankAccount(
         ConsumerPaymentDetails.BankAccount.type,
-        emptyList()
+        emptyList(),
+        R.string.stripe_payment_method_bank,
+        R.drawable.ic_link_bank,
+        primaryButtonStartIconResourceId = R.drawable.ic_link_add
     ) {
-        override fun createParams(
-            paymentMethodCreateParams: PaymentMethodCreateParams,
-            email: String
-        ): ConsumerPaymentDetailsCreateParams {
-            TODO("Not yet implemented")
-        }
+        override fun primaryButtonLabel(
+            stripeIntent: StripeIntent,
+            resources: Resources
+        ) = resources.getString(R.string.add_bank_account)
     };
 
-    /**
-     * Build the [ConsumerPaymentDetailsCreateParams] that will to create this payment method.
-     */
-    abstract fun createParams(
-        paymentMethodCreateParams: PaymentMethodCreateParams,
-        email: String
-    ): ConsumerPaymentDetailsCreateParams
+    val showsForm = formSpec.isNotEmpty()
 
     /**
-     * A map containing additional parameters that must be sent during payment confirmation.
+     * The label for the primary button when this payment method is being created.
      */
-    open fun extraConfirmationParams(paymentMethodCreateParams: PaymentMethodCreateParams):
```

Comment:
Removed some methods from that were here to generalize how we create the different PaymentDetails types. In reality the creation is too different, and these methods were only relevant for Card.